tools/firmware-utils/src/tplink-safeloader.c

```diff
@@ -1562,6 +1562,49 @@ static struct device_info boards[] = {
 		.last_sysupgrade_partition = "file-system"
 	},
 
+  /** Firmware layout for the RE200 v3 */
+	{
+		.id     = "RE200-V3",
+		.vendor = "",
+		.support_list =
+			"SupportList:\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:00000000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:41520000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:41550000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:42520000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:43410000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:45530000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:45550000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:49440000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:4a500000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:4b520000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:52550000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:54570000}\n"
+			"{product_name:RE200,product_ver:3.0.0,special_id:55530000}\n",
```

Comment:
Have you properly extracted these, or just replaced "2.0.0" by "3.0.0" from v2?

Reply:
EU Factory Image contains
```
strings re200v3_eu-up-ver1-1-0-P1\[20190731-rel41250\].bin | grep "special"
{product_name:RE200,product_ver:3.0.0,special_id:00000000}
{product_name:RE200,product_ver:3.0.0,special_id:45550000}
{product_name:RE200,product_ver:3.0.0,special_id:4A500000}
{product_name:RE200,product_ver:3.0.0,special_id:4B520000}
{product_name:RE200,product_ver:3.0.0,special_id:43410000}
{product_name:RE200,product_ver:3.0.0,special_id:41550000}
{product_name:RE200,product_ver:3.0.0,special_id:42520000}
{product_name:RE200,product_ver:3.0.0,special_id:55530000}
{product_name:RE200,product_ver:3.0.0,special_id:41520000}
{product_name:RE200,product_ver:3.0.0,special_id:52550000}
{product_name:RE200,product_ver:3.0.0,special_id:54570000}
{product_name:RE200,product_ver:3.0.0,special_id:45530000}
{product_name:RE200,product_ver:3.0.0,special_id:49440000}
{product_name:RE200,product_ver:3.0.0,special_id:45470000}
```

Reply:
Does order matter? I copied the v2 declarations and compared them to the re200v3_eu-up-ver1-1-0-P1\[20190731-rel41250\].bin. They are the same except order. I can of course use the order coming with the bin file.

Reply:
There were some missing IDs when compared with the v2. Order does not matter for the factory firmware, however i would prepare them to be ordered here.